packages/engine/src/proto.rs

```diff
@@ -2,12 +2,14 @@ use std::fmt::{Debug, Formatter};
 
 use serde::{Deserialize, Serialize};
 use serde_json::Value as SerdeValue;
+use uuid::Uuid;
 
 use crate::{config::Globals, hash_types::worker::RunnerError, simulation::status::SimStatus};
 
 pub type SerdeMap = serde_json::Map<String, SerdeValue>;
 
-pub type ExperimentRegisteredId = String;
+pub type ExperimentName = String;
+pub type ExperimentNameRef<'a> = &'a str;
```

Comment:
I don't know if this is a good idea. It seems a bit messy but we can't just use `&ExperimentName` for `&str` because that's actually `&String`. This causes problems in function signatures. I'm open to swapping to changing signatures to just use `&str` if we prefer.

I considered newtypes and such but all of those are overkill / unergonomic

Reply:
I think this is the best option. My second-best option would be using `&ExperimentName` and changing things so `&String` works. `&str` would keep things ambiguous IMO.

Reply:
Yeah I think we should stick with it as it is. I'm not sure about &String, that's quite an unusual type you don't really want

Reply:
We don't use `ExperimentName` as interchangeable type for `String`, we treat it as a newtype, but with the advantages of `String` like all the trait impls. I'd make it a newtype `struct ExperimentName(String);` and implement `Display` and `FromStr` (for `clap`) and derive the rest (`Debug`, `PartialEq`, `Eq`, `PartialOrd`, `Ord`, `Clone`, `Serialize`, `Deserialize`). If you really want the behavior as `String` just implement `Deref(Mut)`. 
Even if we decide to not go this way, I'd use `&ExperimentName`, as you won't ever write `&String` but `&ExperimentName`. If sticking with the alias or implementing `Deref` it can coerce to `&str`.

Reply:
you also use `&ExperimentId`, It's may also worth converting it to `struct ExperimentId(String);`

Reply:
`ExperimentId` is a `Uuid` not a String so there's type safety.
I've pushed a change making ExperimentName a newtype, it wasn't very ergonomic but I don't think it's too big a problem

Reply:
Yeah, I just remembered. Was still thinking of the "old" `ExperimentId`. We don't use `UUID`s very often so I think it's fine.